src/components/transfers/EditForm.tsx

```diff
@@ -0,0 +1,299 @@
+      validationSchema={validationSchema}>
+      <Box>
+        <Typography variant="h5" component="h2" sx={{ marginBottom: 2, textAlign: 'center' }}>
+          {t('edit-form-heading')}
+        </Typography>
+        <Grid container spacing={2} sx={{ width: 600, margin: '0 auto' }}>
+          <Grid item xs={12}>
+            <FormSelect name="status" label={t('status')}>
+              {(Object.values(TransferStatus) || []).map((status) => {
+                return (
+                  <MenuItem key={status} value={status}>
+                    {status}
+                  </MenuItem>
+                )
+              })}
+            </FormSelect>
+          </Grid>
+          <Grid item xs={12}>
+            <FormTextField type="string" label={t('reason')} name="reason" autoComplete="reason" />
+          </Grid>
+          <Grid item xs={12}>
+            <FormSelect name="currency" label={t('currency')}>
+              {(Object.values(Currency) || []).map((currency) => {
+                return (
+                  <MenuItem key={currency} value={currency}>
+                    {currency}
+                  </MenuItem>
+                )
+              })}
+            </FormSelect>
+          </Grid>
+          <Grid item xs={12}>
+            <FormTextField type="number" label={t('amount')} name="amount" />
+          </Grid>
+          <Grid item xs={12}>
+            <FormDate label={t('targetDate')} name="targetDate"></FormDate>
+          </Grid>
+
+          <Grid item xs={12}>
+            <FormTextField
+              type="string"
+              label={t('documentId')}
+              name="documentId"
+              autoComplete="documentId"
+            />
+          </Grid>
+          <Grid item xs={12}>
+            <FormSelect name="approvedById" label={t('approvedBy')}>
+              {[{ id: '', firstName: <em>None</em>, lastName: '' }, ...people].map((p) => {
+                return (
+                  <MenuItem key={p.id} value={p.id}>
+                    {p.firstName} {p.lastName}
+                  </MenuItem>
+                )
+              })}
+            </FormSelect>
+          </Grid>
+          <Grid item xs={12}>
+            <FormControl size="small" fullWidth>
+              <InputLabel>{t('sourceCampaign')}</InputLabel>
+              <Select
+                type="string"
+                sx={{ height: '55%' }}
+                label={t('sourceCampaign')}
+                name="sourceCampaignId"
+                value={sourceCampaignId}
+                onChange={(event) => {
+                  setSourceVaultId('')
+                  setSourceCampaignId(event.target.value)
+                }}>
+                {[{ id: '', title: '' }, ...campaigns].map((p) => {
+                  return (
+                    <MenuItem key={p.id} value={p.id}>
+                      {p.title}
+                    </MenuItem>
+                  )
+                })}
+              </Select>
+            </FormControl>
```

Comment:
We need a dedicated `SelectCampaign` component and we pass it different name for the source and destination:

```tsx
<>
  <SelectCampaign name="sourceCampaignId" />
  <SelectVault name="sourceCampaignId" />
  <SelectCampaign name="targetCampaignId" />
  <SelectVault name="targetCampaignId" />
</>
```